Modules/OrgUnit/classes/User/class.ilOrgUnitUserRepository.php

```diff
@@ -0,0 +1,215 @@
+use OrgUnit\Positions\ilOrgUnitPosition;
+
+/**
+ * Class ilOrgUnitUserRepository
+ *
+ * @author: Martin Studer   <ms@studer-raimann.ch>
+ */
+class ilOrgUnitUserRepository {
+
+	/**
+	 * @var \ILIAS\DI\Container
+	 */
+	protected $dic;
+	/**
+	 * @var self[]
+	 */
+	protected static $instance;
+	/**
+	 * @var ilOrgUnitUser[]
+	 */
+	protected $orgu_users;
+	/**
+	 * @var bool
+	 */
+	protected $with_superiors = false;
+	/**
+	 * @var bool
+	 */
+	protected $with_positions = false;
+
+
+	/**
+	 * ilOrgUnitUserRepository constructor.
+	 *
+	 */
+	public function __construct() {
+		global $DIC;
+		$this->dic = $DIC;
+	}
+
+
+	/**
+	 * @return ilOrgUnitUserRepository
+	 */
+	public function withSuperiors():ilOrgUnitUserRepository {
+		$this->with_superiors = true;
+
+		return $this;
+	}
+
+
+	/**
+	 * @return ilOrgUnitUserRepository
+	 */
+	public function withPositions():ilOrgUnitUserRepository {
+		$this->with_positions = true;
+
+		return $this;
+	}
+
+
+	/**
+	 * @param array $arr_user_id
+	 *
+	 * @return array
+	 */
+	public function getOrgUnitUsers(array $arr_user_id): array {
+
+		$this->orgu_users = $this->loadUsersByUserIds($arr_user_id);
+
+		if ($this->with_superiors === true) {
+			$this->loadSuperiors($arr_user_id);
+		}
+
+		if ($this->with_positions === true) {
+			$this->loadPositions($arr_user_id);
+		}
+
+		return $this->orgu_users;
+	}
+
+
+	/**
+	 * @param $user_id
+	 *
+	 * @return ilOrgUnitUser|null
+	 */
+	public function getOrgUnitUser($user_id): ?ilOrgUnitUser {
+
+		$this->orgu_users = $this->loadUsersByUserIds([ $user_id ]);
+
+		if (count($this->orgu_users) == 0) {
+			return null;
+		}
+
+		if ($this->with_superiors === true) {
+			$this->loadSuperiors([ $user_id ]);
+		}
+
+		return $this->orgu_users[0];
+	}
+
+
+	/**
+	 * @param array $user_ids
+	 */
+	public function loadSuperiors(array $user_ids): void {
+
+		global $DIC;
+
+		$st = $DIC->database()->query($this->getSuperiorsSql($user_ids));
+
+		$empl_id_sup_ids = [];
+		while ($data = $DIC->database()->fetchAssoc($st)) {
+			$org_unit_user = ilOrgUnitUser::getInstanceById($data['empl_usr_id']);
+			$superior = ilOrgUnitUser::getInstance($data['sup_usr_id'], $data['sup_login'], $data['sup_email']);
+			$org_unit_user->addSuperior($superior);
+		}
+	}
+
+
+	/**
+	 * @param array $user_ids
+	 *
+	 * @return array
+	 */
+	public function getEmailAdressesOfSuperiors(array $user_ids): array {
+
+		global $DIC;
+
+		$st = $DIC->database()->query($this->getSuperiorsSql($user_ids));
+
+		$arr_email_sup = [];
+		while ($data = $DIC->database()->fetchAssoc($st)) {
+			$arr_email_sup[] = $data['sup_email'];
+		}
+
+		return $arr_email_sup;
+	}
+
+
+	/**
+	 * @param array $user_ids
+	 *
+	 * @return string
+	 */
+	protected function getSuperiorsSql(array $user_ids):string {
+		global $DIC;
+
+		$sql = "SELECT 
+				orgu_ua.orgu_id AS orgu_id,
+				orgu_ua.user_id AS empl_usr_id,
+				orgu_ua2.user_id as sup_usr_id,
+				superior.email as sup_email,
```

Comment:
Please keep in mind that a user record also has a `second_email`. I personally don't need this for my current ILIAS 6.0.x feature (1st consumer of this OrgUnit API), but maybe we should address this in general.